PythonTool/TagExpansion.py

```diff
@@ -168,20 +178,42 @@ def expand_vector_tags(mesh_file, main_dir_name = None):
        LookupError: If the file does not contain any vector tags.
     """
 
+    # Create a dictionary with the MB element types and their integer values.
+    elements = {
+        "vertex" : 0, "edge" : 1, "tri" : 2, "quad" : 3, "polygon" : 4,
+        "tet" : 5, "pyramid" : 6, "prism" : 7, "knife" : 8, "hex" : 9,
+        "polyhedron" : 10, "entityset" : 11, "maxtype" : 12
+        }
+
     # Load the mesh file to create a reference mesh.
     mb_ref = core.Core()
     mb_ref.load_file(mesh_file)
 
-    # Retrieve the lists of scalar and vector tags on the mesh.
-    try:
-        hexes_ref, scal_tags_ref, vec_tags_ref = get_tag_lists(mb_ref, types.MBHEX)
-    except LookupError as e:
-        print(str(e))
-        exit()
+    # Retrieve the lists of scalar and vector tags on the reference mesh.
+    if element_type is None:
+        # Search for hex elements in the mesh.
+        element_type = "hex"
+        try:
+            hexes_ref, scal_tags_ref, vec_tags_ref = get_tag_lists(mb_ref, element_type, types.MBHEX)
+        except LookupError as e:
+            print(str(e))
+            exit()
+    else:
+        # Ensure the user entered a valid MB element type.
+        type_int = elements.get(element_type.lower())
+        if type_int is None:
+            print("WARNING: Please choose a valid MB element type.")
+            exit()
+        else:
+            try:
+                hexes_ref, scal_tags_ref, vec_tags_ref = get_tag_lists(mb_ref, element_type, type_int)
+            except LookupError as e:
```

Comment:
I'll echo Kalin's comments in a previous PR that this can be written as: 

```
raise e("error message")
```

Reply:
This is actually there to catch an exception that would be thrown from the `get_tag_lists` function! Which is called in the body of the `try` block. So the error itself is raised elsewhere and needs to be caught here instead.